zipline/assets/roll_finder.py

```diff
@@ -179,20 +179,20 @@ def _active_contract(self, oc, front, back, dt):
         front_contract = oc.sid_to_contract[front].contract
         back_contract = oc.sid_to_contract[back].contract
 
+        tc = self.trading_calendar
+        trading_day = tc.day
+        prev = dt - trading_day
+        get_value = self.session_reader.get_value
```

Comment:
These four lines that moved are, but below I had to change `dt` to `prev` because `prev` is the actual day that we are calling `get_value` on.